apiserver/facades/controller/firewaller/firewaller.go

```diff
@@ -767,3 +767,33 @@ func (f *FirewallerAPIV6) GetExposeInfo(args params.Entities) (params.ExposeInfo
 	}
 	return result, nil
 }
+
+// SpaceInfos returns a comprehensive representation of either all spaces or
+// a filtered subset of the known spaces and their associated subnet details.
+func (f *FirewallerAPIV6) SpaceInfos(args params.SpaceInfosParams) (params.SpaceInfos, error) {
```

Comment:
@manadart This is the one we were talking about. All the marshal/unmarshal logic is encapsulated in the params package so we can easily clone this into any other facades that need this functionality.